File: QuasselDroidNG/src/main/java/de/kuschku/libquassel/primitives/serializers/IntSerializer.java
package de.kuschku.libquassel.primitives.serializers;

import android.support.annotation.NonNull;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.channels.ByteChannel;

public class IntSerializer implements PrimitiveSerializer<Integer> {
    @NonNull
    private static final IntSerializer serializer = new IntSerializer();

    private IntSerializer() {
    }

    @NonNull
    public static IntSerializer get() {
        return serializer;
    }

    @Override
    public void serialize(@NonNull final ByteChannel channel, @NonNull final Integer data) throws IOException {
        final ByteBuffer buffer = ByteBuffer.allocate(4);
        buffer.putInt(0, data);
        channel.write(buffer);
    }

    @NonNull
    @Override
    public Integer deserialize(@NonNull final ByteBuffer buffer) throws IOException {
        return buffer.getInt();
    }
}

File: app/src/main/java/de/kuschku/quasseldroid_ng/protocol/primitive/serializer/IntSerializer.kt
package de.kuschku.quasseldroid_ng.protocol.primitive.serializer

import de.kuschku.quasseldroid_ng.protocol.Quassel_Features
import de.kuschku.quasseldroid_ng.util.nio.ChainedByteBuffer
import java.nio.ByteBuffer

object IntSerializer : Serializer<Int> {
  override fun serialize(buffer: ChainedByteBuffer, data: Int, features: Quassel_Features) {
    buffer.putInt(data)
  }

  override fun deserialize(buffer: ByteBuffer, features: Quassel_Features): Int {
    return buffer.int
  }
}
